### ISSUE TITLE
VSCode does not honor .gitignore file if the project has no .git directory

### ISSUE THREAD

**Issue by myelsukov** [Microsoft/vscode#69231]
Issue Type: <b>Bug</b>

Remove `.git` directory or workspace's `.git`  file from the project.

Result: VSCode stops honoring the `.gitignore` file.

Expectation: VSCode honors `.gitignore` file irrespective of the presense of the `.git` directory or file. 

Many other applications (e.g. `rsync`) honor `.gitignore` file no matter what :)

VS Code version: Code 1.31.1 (1b8e8302e405050205e69b59abb3559592bb9e60, 2019-02-12T02:16:38.656Z)
OS version: Darwin x64 17.7.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-7920HQ CPU @ 3.10GHz (8 x 3100)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|2, 2, 2|
|Memory (System)|16.00GB (0.82GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (24)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-direnv|Rub|0.0.2
numbered-bookmarks|ale|6.1.0
bracket-pair-colorizer|Coe|1.0.61
vscode-eslint|dba|1.8.0
githistory|don|0.4.6
gitlens|eam|9.5.1
tslint|eg2|1.0.43
prettier-vscode|esb|1.8.1
git-project-manager|fel|1.7.1
code-runner|for|0.9.7
vscode-pull-request-github|Git|0.4.0
githd|hui|2.1.0
rest-client|hum|0.21.1
asciidoctor-vscode|joa|2.3.3
my-keymap|MY|0.0.1
vscode-extension-auto-import|Nuc|1.4.3
indent-rainbow|ode|7.2.4
vscode-yaml|red|0.3.0
partial-diff|ryu|1.4.0
code-settings-sync|Sha|3.2.5
incremental-search|sie|0.2.1
html-preview-vscode|tht|0.2.5
json2yaml|tux|0.2.0
vscode-surround|yat|1.0.2

</details>

**Comment by jrieken** [comment]
No, the design is to shell out to git for checking what's ignored, see https://git-scm.com/docs/git-check-ignore, because only that provides the true picture of what's being ignore by git.

**Comment by myelsukov** [comment]
I expect VSCode to ignore files and directories mentioned in `.gitignore` irrespective of the presence or absence of the `.git` directory. For example, it shouldn't search for files in those directories/files.

There are many well known utilities that understand and use `.gitignore` files to do the filtering. One of them is `rsync` which does not required `git` to be installed.

I understand why it could be hard to implement in VSCode, but I would called it `as-implemented`, not `as-designed` :)

**Comment by jrieken** [comment]
> For example, it shouldn't search for files in those directories/files.

It does that, but for greying out files in the explorer is consults git as the source of truth.

**Comment by jrieken** [comment]
> I understand why it could be hard to implement in VSCode,

This isn't about that. The problem would be the false impression, e.g you add a file to .gitignore, the explorer shows the file as ignored (greyed) but subsequent edits in that file show as modification because it's actually not ignored.

**Comment by myelsukov** [comment]
> It does that

This is how I discovered the "problem". I made a copy of my project (don't ask me about reasons :) to do some work. Obviously I didn't copy the `.git`. After that search in files stopped ignoring directories mentioned in `.gitignore` so I got a lot of matches where I expected just a few of them. Frankly, this is the biggest inconvenience to me.

> but subsequent edits in that file show as modification because it's actually not ignored.

We should define what does it mean to be 'ignored' :) To me, ignored means that the file is not considered to be a part of the project. It just resides there among other files (but ignored). In a sense, `.git` directory is not a part of the project. This is just how git stores things. That's why VSCode doesn't search in the `.git` directory.
